### PR TITLE
Make string methods return a Series with a useful Index

### PR DESCRIPTION
**Pull request by shwina**
## Description
Closes https://github.com/rapidsai/cudf/issues/12806

Many string methods like `character_ngrams` currently return a `Series` with the default index (`RangeIndex`). This PR makes it so that the index of the result corresponds to the index of the input.

More specifically, this PR changes the index of the result of the following string methods:

- [x] `character_ngrams`
- [x] `tokenize`
- [x] `character_tokenize`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
@shwina was this PR intentionally marked as draft?

**Comment by shwina** [comment]
@vyasr - should be ready for review now

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/core/column/string.py [hunk 1]
```diff
@@ -4641,41 +4646,43 @@ def character_tokenize(self) -> SeriesOrIndex:
...
+        2    y
+        2    o
+        2    u
+        2    .
         dtype: object
         """
         result_col = libstrings.character_tokenize(self._column)
         if isinstance(self._parent, cudf.Series):
-            return cudf.Series(result_col, name=self._parent.name)
+            lengths = self.len().fillna(0)
```
Comment (vyasr) [inline python/cudf/cudf/core/column/string.py:4683]: Is the `fillna(0)` matching pandas behavior? It looks like the test is constructing the expected output manually to be the same, so it's not clear if it's going to match pandas or if the test is just constructing the same as what this outputs.

**Review by vyasr — APPROVED** [review]
One question about null behavior but I'm approving and will let you resolve accordingly.

**Pull request merged**